### DISCUSSION TITLE
Idea 1

### DISCUSSION THREAD

**Discussion by Tebarem** [letsbuilda/slayer-in-the-shadows#1] [Ideas]
a secret organization that has been fighting in the shadows for centuries. The player character used to be a member of the organization till he got betrayed(more on that later)

The game loop is you advance through the levels aka the building of the org floor by floor looking for the boss as he could be anywhere.

Mechanics you can dash, double-jump, slowdown time, slice and dice, stealth kills(head twists) and throw knives all that being chainable events meaning you are able to dash first if u hit a enemy you have the option to slice and dice, stealth kill or throw a knife, if you kill the enemy you are able to dash out or double jump again e.t.c

we can kind of make the enemies fade away
or flash the screen when the enemy dies
i can see it being nice when the screen flashes and then they just disappear in a smoke
would be cool to chain attacks while the screen is flashing as well

So would the player be trying to take out as many people from the org as possible, or is it just the boss specifically?
we can make it so that in each floor it might be someone from the org
that u know that betrayed you etc

I think the user should get bonus points or smth for not taking out other people then, since that could raise suspicion from the other people and be less stealthy
Other way would be getting a score/stars, and you get 3 for not being seen/heard, 2 if you were seen but lived, 1 if you killed someone who wasn't the target, and none if you die

i think we should have different difficulties
for example easy lets u have a health bar
where u can fight them
normal as soon as ur hit or touched u lose
and expert as soon as ur seen u lose

**Comment by Tebarem** [comment]
So how do we start this...

**Reply by Tebarem** [reply]
Movement System: This system would handle the player's movement, including dashing, double-jumping, and moving around the game environment. It would also handle collision detection and physics, ensuring that the player moves smoothly and fluidly through the game world.

Combat System: The combat system would handle the player's attacks and defensive moves, including slicing and dicing, stealth kills, and throwing knives. It would also handle enemy AI and hit detection, ensuring that attacks land and deal damage to enemies when they are supposed to.

Time-Slowdown System: This system would handle the player's ability to slow down time, allowing them to move more quickly and precisely during combat. It would also handle the visual effects associated with time slowdown, including slow-motion animations and visual distortions.

Chainable Moves System: This system would allow the player to chain together different moves and combos during combat. It would handle the logic behind move chaining, allowing the player to seamlessly transition from one move to the next based on user input and context.

Level Design System: This system would handle the creation of levels and game environments, including the placement of enemies, obstacles, and other game elements. It would also handle level loading and transition logic, ensuring that the player moves smoothly between levels as they progress through the game.

**Reply by Tebarem** [reply]
Enemy AI System: This system would handle the behavior and movement of enemy characters in the game. It would include logic for enemy pathfinding, awareness, and combat behavior. For example, enemies might move in patterns or patrol areas, become alerted when they hear a noise or see the player, and engage the player in combat when they get close enough. The system would also handle enemy attacks, including melee attacks and ranged attacks like throwing knives, and damage detection.

Enemy Health and Damage System: This system would handle enemy health and damage, including the logic for determining when an enemy is defeated and how much damage different player attacks do to different types of enemies.

**Comment by Anonymous4045** [comment]
Thinking back, it might be more fun if this is more of a combat-style game than a stealth game. There'd still be a lot of sneaking around, but the player's goal is instead to take out as many people from the bad organization as possible. That way we'd put the combat system you talked about to good use.

**Reply by Tebarem** [reply]
Hmmm. What if the more you stay stealthy the more you can chain your attacks. For example instead of being able to "Attack reset" on every kill. If you stayed stealthy enough for lets say 30 seconds, you can chain attack a single enemy till they die.

**Reply by Aurabiq** [reply]
We should add enemy boss that's hard to kill

**Reply by Anonymous4045** [reply]
Hmm, that could work, but wouldn't make too much sense imo. Maybe the enemies have some sort of path pattern they follow, and once they turn their back after a bit, the player can sneak in and get more hits in before the enemy realizes.

**Reply by Tebarem** [reply]
> Hmm, that could work, but wouldn't make too much sense imo. Maybe the enemies have some sort of path pattern they follow, and once they turn their back after a bit, the player can sneak in and get more hits in before the enemy realizes.

This sounds good and not too complicated to do. But would be nice to have variations of enemies.

**Reply by Tebarem** [reply]
But final boss in a stealthy game i dont know how that would work. Instead i think should be a sort of assasination or something. What if we just do a cliff hanger. The boss lives and the building explodes leaving the player dead.

**Reply by Anonymous4045** [reply]
> would be nice to have variations of enemies.

Yeah, there can be some faster, some slower, some with ranged weapons, etc. Maybe they could also drop weapons the player can pick up?

**Reply by Tebarem** [reply]
>  Maybe they could also drop weapons the player can pick up?

What if we do something like doom. Instead of dropping weapons they drop ammo.

**Reply by Aurabiq** [reply]
but what about game **art**

**Reply by Tebarem** [reply]
> but what about game **art**

we can find assets online and modify them. We also have robin.

**Reply by Anonymous4045** [reply]
> Instead of dropping weapons they drop ammo

That could work, but then the player would need to have inventory slots for the various weapons. I think it might be more ninja-like if they only had one weapon at a time, to be quieter.

Different weapons could be like this:
Dagger: attacks quickly but deals smaller damage.
Katana: medium attack speed and damage.
Longsword: high damage but slow speed. If we want to make sound be a thing, this would be the loudest and make you easily detected.
Bow: requires arrows around the map or dropped from enemies.
Composite bow: requires arrows, takes longer to pull back but does more damage than the normal bow.
Shuriken: secondary throwable item found around the map, doesn't count as primary weapon.

I think it'd be game jam-y if the player can only have one primary weapon at a time, obtained from defeating enemies carrying the weapon.

**Reply by Aurabiq** [reply]
> > Instead of dropping weapons they drop ammo
> 
> That could work, but then the player would need to have inventory slots for the various weapons. I think it might be more ninja-like if they only had one weapon at a time, to be quieter.
> 
> Different weapons could be like this: Dagger: attacks quickly but deals smaller damage. Katana: medium attack speed and damage. Longsword: high damage but slow speed. If we want to make sound be a thing, this would be the loudest and make you easily detected. Bow: requires arrows around the map or dropped from enemies. Composite bow: requires arrows, takes longer to pull back but does more damage than the normal bow. Shuriken: secondary throwable item found around the map, doesn't count as primary weapon.
> 
> I think it'd be game jam-y if the player can only have one primary weapon at a time, obtained from defeating enemies carrying the weapon.

### Hmm

**Reply by Tebarem** [reply]
> > Instead of dropping weapons they drop ammo
> 
> That could work, but then the player would need to have inventory slots for the various weapons. I think it might be more ninja-like if they only had one weapon at a time, to be quieter.
> 
> Different weapons could be like this: Dagger: attacks quickly but deals smaller damage. Katana: medium attack speed and damage. Longsword: high damage but slow speed. If we want to make sound be a thing, this would be the loudest and make you easily detected. Bow: requires arrows around the map or dropped from enemies. Composite bow: requires arrows, takes longer to pull back but does more damage than the normal bow. Shuriken: secondary throwable item found around the map, doesn't count as primary weapon.
> 
> I think it'd be game jam-y if the player can only have one primary weapon at a time, obtained from defeating enemies carrying the weapon.

this might be overkill(?) imo. We should keep it simple. Dagger/sword/katana (they all do samething), and throwing knife instead of bow

**Reply by Anonymous4045** [reply]
> We should keep it simple

Alright, do you think the player should be able to get all those weapons at once in their inventory though, or should they only have one at a time?

**Reply by Tebarem** [reply]
> or should they only have one at a time?

should be only 1 melee and 1 range.

**Reply by Aurabiq** [reply]
now I'm back

**Reply by FieryIceStickie** [reply]
i like the idea of just having throwing knives for range and katana for melee and that's it
and then focus more on movement options to expand the player's abilities

**Reply by henri-climber** [reply]
> But final boss in a stealthy game i dont know how that would work. Instead i think should be a sort of assasination or something. What if we just do a cliff hanger. The boss lives and the building explodes leaving the player dead.

Yea Final Boss doesn't make much sense imo as well. 
What do you think about having sth like a small riddle in the last level? Maybe a countdown and if you don't solve the riddle everything explodes/you die and lose

**Reply by FieryIceStickie** [reply]
Maybe the last level could go back through all the previous levels with slight modifications and something new hidden somewhere in the level that you would figure out through the riddles

**Comment by Aurabiq** [comment]
do you start coding?

**Reply by Anonymous4045** [reply]
Only when we start making issues and assigning them.

**Comment by henri-climber** [comment]
I like the idea so far, sounds cool.  

Do you want to make the screen move both horizontally and vertically? Or was the idea to make each floor be a completely separate level?

**Reply by FieryIceStickie** [reply]
A side-scroller going horizontally through all the rooms of a level seem good to me
And like once you move out of one of the sides of a room it moves to the next room in the level
Making it go vertical too seems like too much level to design in a week

**Reply by Tebarem** [reply]
> A side-scroller going horizontally through all the rooms of a level seem good

now that i think about it... it might make sense to set this in some dojo environement rather than a tower...

**Reply by Tebarem** [reply]
or actually just have the tower be one big dojo

**Reply by Anonymous4045** [reply]
If it's to be a side-scroller, what do you guys think of the camera auto-scrolling and it crushes you if you don't move fast enough? Adds a time limit to each level.

**Reply by brodycritchlow** [reply]
> If it's to be a side-scroller, what do you guys think of the camera auto-scrolling and it crushes you if you don't move fast enough? Adds a time limit to each level.

I am down for that idea, would add another detail to the game.

**Reply by henri-climber** [reply]
> If it's to be a side-scroller, what do you guys think of the camera auto-scrolling and it crushes you if you don't move fast enough? Adds a time limit to each level.

yea that sounds cool however, that will make it harder to implement shadow features because you can't hide for longer times. Maybe we need to figure out how to get both features implemented nicely

**Reply by Anonymous4045** [reply]
> we need to figure out how to get both features implemented nicely

How about this? The player fights through the levels at their own pace until they reach the end and fight a boss or something. Then, once everyone from the org is dead, the building/tower thing is set to self-destruct, and the screen starts to scroll back to the left. The player must escape as the building is collapsing behind them.

**Comment by Robin5605** [comment]
I'm liking all the ideas! Keep them flowing.

**Comment by asterized** [comment]
idea which is an idea
